orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.java

```diff
@@ -0,0 +1,649 @@
+import static com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType.ORCHESTRATION;
+import static com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType.PIPELINE;
+import static com.netflix.spinnaker.orca.pipeline.model.SyntheticStageOwner.STAGE_BEFORE;
+import static java.lang.System.currentTimeMillis;
+import static java.lang.String.format;
+import static java.util.Collections.*;
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static redis.clients.jedis.BinaryClient.LIST_POSITION.AFTER;
+import static redis.clients.jedis.BinaryClient.LIST_POSITION.BEFORE;
+
+@Component
+@Slf4j
+public class JedisExecutionRepository extends AbstractRedisExecutionRepository {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  @Autowired
+  public JedisExecutionRepository(
+    Registry registry,
+    @Qualifier("redisClientDelegate") RedisClientDelegate redisClientDelegate,
+    @Qualifier("previousRedisClientDelegate") Optional<RedisClientDelegate> previousRedisClientDelegate,
+    @Qualifier("queryAllScheduler") Scheduler queryAllScheduler,
+    @Qualifier("queryByAppScheduler") Scheduler queryByAppScheduler,
+    @Value("${chunkSize.executionRepository:75}") Integer threadPoolChunkSize
+  ) {
+    super(registry, redisClientDelegate, previousRedisClientDelegate, queryAllScheduler, queryByAppScheduler, threadPoolChunkSize);
+  }
+
+  public JedisExecutionRepository(
+    Registry registry,
+    RedisClientDelegate redisClientDelegate,
+    Optional<RedisClientDelegate> previousRedisClientDelegate,
+    Integer threadPoolSize,
+    Integer threadPoolChunkSize
+  ) {
+    super(
+      registry,
+      redisClientDelegate,
+      previousRedisClientDelegate,
+      Schedulers.from(Executors.newFixedThreadPool(10)),
+      Schedulers.from(Executors.newFixedThreadPool(threadPoolSize)),
+      threadPoolChunkSize
+    );
+  }
+
+  @Override
+  public void store(@Nonnull Execution execution) {
+    getRedisDelegateForId(execution.getId()).withTransaction(tx -> {
+      storeExecutionInternal(tx, execution);
+      if (execution.getType() == PIPELINE) {
+        tx.zadd(executionsByPipelineKey(
+          execution.getPipelineConfigId()),
+          execution.getBuildTime() != null ? execution.getBuildTime() : currentTimeMillis(),
+          execution.getId());
+      }
+      tx.exec();
+    });
+  }
+
+  @Override
+  public void storeStage(@Nonnull Stage stage) {
+    ExecutionType type = stage.getExecution().getType();
+    String key = format("%s:%s", type.toString(), stage.getExecution().getId());
+    getRedisDelegateForId(key).withTransaction(tx -> {
+      storeStageInternal(tx, type, stage);
+      tx.exec();
+    });
+  }
+
+  @Override
+  public void removeStage(@Nonnull Execution execution, @Nonnull String stageId) {
+    String key = format("%s:%s", execution.getType().toString(), execution.getId());
```

Comment:
You shouldn't need to specify `.toString()` in these format expressions. The `%s` in the expression does that anyway.

Reply:
👍